src/hotspot/share/runtime/frame.cpp

```diff
@@ -205,11 +205,11 @@ void RegisterMap::print() const {
 
 address frame::raw_pc() const {
   if (is_deoptimized_frame()) {
-    CompiledMethod* cm = cb()->as_compiled_method_or_null();
-    if (cm->is_method_handle_return(pc()))
-      return cm->deopt_mh_handler_begin() - pc_return_offset;
+    nmethod* nm = cb()->as_nmethod_or_null();
```

Comment:
Prexisting: It's weird that this code is using the `_or_null()` version when the code below does not null check the returned value.

Reply:
Before [JDK-6921352](https://bugs.openjdk.org/browse/JDK-6921352) it was:
```
  return ((nmethod*) cb())->deopt_handler_begin() - pc_return_offset;
```

I will add assert with check for null. We definitely expect here only nmethod.